### ISSUE TITLE
Compression vision encoders 

### ISSUE THREAD

**Issue by youssefmohana** [IPMI-NWU/GD-ViG#3]
I tried to train vision encoder images only can you share code compression where my code gives accuracy 50 percent any help please code or experiments details please (training in eye gaze cxr )

**Comment by SX-SS** [comment]
Thanks for your interest in this work. 
The possible reason for the low accuracy is that you are perhaps not using a pre-trained model from Vision GNN. 
Pre-training parameters for Vision GNN on ImageNet are available at this link: https://github.com/huawei-noah/Efficient-AI-Backbones/tree/master/vig_pytorch. Thanks for the outstanding contribution to this paper: "Vision GNN: An Image is Worth Graph of Nodes".
Then you can initialise your encoder in a simple way like this:
```
self.Encoder = ViG()
save_model = torch.load(‘pvig_s_82.1.pth.tar’)
model_dict = self.Encoder.state_dict()
state_dict = {k: v for k, v in save_model.items() if k in model_dict.keys()}
model_dict.update(state_dict)
self.Encoder.load_state_dict(model_dict)
```
Such a transfer learning training approach makes GNN models more stable.

**Comment by youssefmohana** [comment]
Thanks for help  i use weights from pvig_ti before in my experiments so I think this make results low (paper implementation use vig encoder (pvig ti ) structure )   and when load model give error where in unet in paper changed position encoding 224 / 4 , 224/4 (56 , 56 )

**Comment by youssefmohana** [comment]
Can i ask type of classifier with backbone compression where resnet , swin and etc
